crates/services/txpool_v2/src/extracted_outputs.rs

```diff
@@ -0,0 +1,129 @@
+//! Temporary module to extract outputs until we know that they have been settled in database or skipped.
+
+use std::collections::HashMap;
+
+use fuel_core_types::{
+    fuel_tx::{
+        input::coin::{
+            CoinPredicate,
+            CoinSigned,
+        },
+        Address,
+        AssetId,
+        ContractId,
+        Input,
+        Output,
+        TxId,
+        UtxoId,
+    },
+    services::txpool::ArcPoolTx,
+};
+
+pub struct ExtractedOutputs {
+    contract_created: HashMap<ContractId, TxId>,
+    contract_created_by_tx: HashMap<TxId, Vec<ContractId>>,
+    coins_created: HashMap<TxId, HashMap<u16, (Address, u64, AssetId)>>,
```

Comment:
A nit:
For clarity, `HashMap<TxId, HashMap<u16, (Address, u64, AssetId)>>` maybe deserves a dedicated type to be introduced.

It'll also allow to make [code like this](https://github.com/FuelLabs/fuel-core/pull/2855/files#diff-a5b4c3f0c7a4bf653cd9d7e88b8aa121c8b966395ef1c89823ff99638dd5e67aR124-R126) more lean: `a == address && am == amount && asid == asset_id`